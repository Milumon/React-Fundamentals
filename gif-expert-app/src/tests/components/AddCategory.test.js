import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Test de AddCategory', () => {

    const setCategories = () => {}
    const wrapper = shallow( <AddCategory setCategories={setCategories} /> )

    test('Debe de mostrarse el component', () => {

         expect(wrapper).toMatchSnapshot();
        
    });

    test('Debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: {value} } )

         expect( wrapper.find('h1').text().trim() ).toBe(value)
        
    })
    
    
});