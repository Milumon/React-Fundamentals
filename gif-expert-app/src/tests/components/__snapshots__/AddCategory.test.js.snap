// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de AddCategory Debe de mostrarse el component 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
